packages/core/datadog-sourcemaps.gradle

```diff
@@ -0,0 +1,139 @@
+        def targetPath = variant.dirName
+
+        def bundleTask = tasks.findByName("bundle${targetName}JsAndAssets")
+
+        if (bundleTask == null) {
+            logger.info("Cannot find JS bundle task for target=${targetName}.")
+            return
+        }
+        if (!bundleTask.enabled) {
+            logger.info("JS bundle task for target=${targetName} is not enabled.")
+            return
+        }
+
+        def bundleAssetName = reactConfig.bundleAssetName ?: "index.android.bundle"
+
+        def jsBundleDir = file("$buildDir/generated/assets/react/${targetPath}")
+        def jsBundleFile = file("$jsBundleDir/$bundleAssetName")
+
+        def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")
+        def jsOutputSourceMapFile = file("$jsSourceMapsDir/${bundleAssetName}.map")
+
+        def uploadTask = tasks.create("upload${targetName}Sourcemaps") {
+            group = "datadog"
+            description = "Uploads sourcemaps to Datadog."
+
+            def execCommand = [
+                    "yarn",
+                    "datadog-ci",
+                    "react-native",
+                    "upload",
+                    "--platform",
+                    "android",
+                    "--service",
+                    serviceName,
+                    "--bundle",
+                    jsBundleFile.absolutePath,
+                    "--sourcemap",
+                    jsOutputSourceMapFile.absolutePath,
+                    "--release-version",
+                    releaseVersion,
+                    "--build-version",
+                    buildVersion
+            ]
+
+            doFirst {
+                if (!jsBundleFile.exists()) {
+                    throw new GradleException("JS bundle file doesn't exist, aborting upload.")
+                }
+
+                if (!jsOutputSourceMapFile.exists()) {
+                    throw new GradleException("JS sourcemap file doesn't exist, aborting upload.")
+                }
+
+                def env = [:]
+                if (System.getenv("DATADOG_API_KEY") == null) {
+                    def keyFile = new File(reactRoot, "datadog.credentials")
+                    if (!keyFile.exists() || keyFile.text.isEmpty()) {
+                        throw new GradleException("Cannot find Datadog API key, it is neither" +
+                                " in env, nor in datadog.credentials file.")
+                    }
+                    env["DATADOG_API_KEY"] = keyFile.text
+                }
+
+                logger.info("Going to run sourcemaps upload using the following command:" +
+                        " ${execCommand.join(" ")} and env=${env}")
+                runShellCommand(execCommand, env)
+
+            }
+        }
+
+        uploadTask.dependsOn bundleTask
+        bundleTask.finalizedBy uploadTask
+
+    }
+}
+
+def runShellCommand(List<String> command, Map<String, String> env = [:]) {
```

Comment:
there's already an `Exec` gradle task from which you can inherit one. Here's the example of the one I did when working for the nightly tests coverage:
```
tasks.create("checkNightlyTestsCoverage", Exec::class.java) {
        this.group = "datadog"
        val generateApiTasks = allprojects.mapNotNull { it.tasks.findByName("generateApiSurface") }
        setDependsOn(generateApiTasks)
        setCommandLine(
            "python3",
            "instrumentation-tools/nightly_tests_code_coverage.py",
            "-t $threshold"
        )
    }
```

Reply:
yes, inside `runShellCommand` there is a call to `exec`, just the body of the upload task created is responsible for the configuration phase, and actual upload is triggered in the `doFirst`.

Reply:
Yup, sorry didn't check your last commit ;)